pkg/auth/utils.go

```diff
@@ -62,3 +64,24 @@ func GetJWTExpiry(token string) (time.Time, error) {
 
 	return claims.Expiry.Time(), nil
 }
+
+// VerifyJWTFromRequest will verify the incoming request contains the correct JWT token
+func VerifyJWTFromRequest(ctx context.Context, tokenVerifier *OIDCTokenVerifier, r *http.Request, audience string, response http.ResponseWriter) (http.ResponseWriter, error) {
+	token := GetJWTFromHeader(r.Header)
+	if token == "" {
+		response.WriteHeader(nethttp.StatusUnauthorized)
+		return response, fmt.Errorf("no JWT token found in request")
+	}
+
+	if audience == "" {
+		response.WriteHeader(nethttp.StatusInternalServerError)
+		return response, fmt.Errorf("no audience is provided")
+	}
+
+	if _, err := tokenVerifier.VerifyJWT(ctx, token, audience); err != nil {
+		response.WriteHeader(http.StatusUnauthorized)
+		return response, fmt.Errorf("failed to verify JWT: %w", err)
+	}
+
+	return response, nil
```

Comment:
not sure, if we need to return the writer here again

Reply:
Tested, not really. Changes applied. Thanks for catching this!